Data Products (#745)
snowplow/documentation: docs/understanding-tracking-design/managing-event-specifications/ui/index.md

  offerings:
    - enterprise
    - cloud
---

## Crafting a New Event Specification through the Console

Event specifications serve as direct counterparts to [data structures](/docs/understanding-tracking-design/managing-your-data-structures/ui/index.md) and encapsulate the documented events within a data product.

:::info
Please note that the creation of event specifications is exclusive to data products.
:::

Create event specifications in tandem with the latest deployed data structure version in production. This ensures tracking implementation instructions align with the validation criteria in the event specification.

Your event specification guarantees compatibility with the associated data structure version. This ensures events tracked using specified instructions pass validation for the associated data structure.

If a newer event specification version is deployed, consider upgrading event specification and tracking to the latest data structure, ensuring ongoing compliance with validation requirements.

## Creating and Editing Event Specifications

To create a new event specification, follow these steps:

1. Select a data product
2. Click the "Create event" button

> 3. A modal will appear, allowing you to enter the event name

Reviewer: ```suggestion
3. A modal will appear, prompting you to enter a name for your event specification and click "Save and continue"
```

4. Your first event specification will be displayed on the page

> ![Create an Event Specification](images/create-event-specification.png)

Reviewer: These screenshots should be replaced with the updated copy changes I think

Author: done

To add more information or modify an existing event specification, follow these steps:

1. Navigate to the appropriate data product
2. Select the desired event specification

> 3. This action will open a page displaying the details of the selected event specification

Reviewer: ```suggestion
3. This action will open on overview of the selected event specification containing the details that have been added to date.
```

> This page is divided into different sections; below, you'll find more details for each section.

Reviewer: ```suggestion
This interface is divided into focused sections; explore each section below for more details.
```

> ![Example of an Event Specification Overview](images/event-specification-overvioew.png)

Reviewer: Would be good to update this image too in line with the recent copy changes

Author: done thanks

### Event Information

This section provides essential meta-information for your event specification, including the event name, description, and the applications in which this event is tracked.

### Event Data Structure

This section defines the structure of your data and specifies the types of properties associated with this event, along with its version.

You can choose from two types of data structures:

> - **Standard**; Provided by the SnowPlow tracker by default

Reviewer: ```suggestion
- **Standard**; Provided by the Snowplow tracker by default
```

- **Custom**; Provided by your organization

![Event Data Structure](images/event-data-structure.png)

## Entity Data Structures

Within this section, you have the flexibility to choose the specific entities that should be associated with the event when it is triggered.

> Moreover, the section not only allows you to select entities but also provides comprehensive details about the properties associated with these entities and their respective versions.

Reviewer: ```suggestion
To facilitate making an informed selection of your entities, you are able to view a detailed breakdown of the properties associated with your chosen entity for a specific version.
```

![Entity Data Structures](images/entity-data-structures.png)

### Event Triggers

This section defines the locations and circumstances under which this event is triggered.

> To create your first trigger, simply select the "Add trigger" button. To edit an existing trigger, click on the corresponding item in the trigger list.

Reviewer: ```suggestion
To create your first trigger, simply select the "Add trigger" button. You can edit or delete an existing trigger by clicking the dedicated buttons beside each entry in the triggers list.
```

A modal will appear, allowing you to upload an image and provide additional context, such as the URL to the page on which this trigger applies.

![Event Triggers](images/event-triggers.png)

> ### Implementation Instructions

Reviewer: ```suggestion
### Properties
```

> The Implementation Instructions section allows you to specify how each property for a selected data structure should be populated.

Reviewer: ```suggestion
This section allows you to specify how each property for a selected event data structure should be populated.
```

> The dialog displays the list of properties defined in the data structure associated with this event specification.

Reviewer: We seem to use both `modal` and `dialog` in different places. Would it make more sense to stick with `dialog` as maybe that is more commonly understood?

Author: done, from modal to dialog, included in this PR only changes in the data products section, in a separate PR I will replace all over modal text, thanks for the suggestion

> Click on any of the properties shown in the list to access a new view inside the dialog box. Here, you can configure the instructions for the Type of Condition, Value, and Comments properties.

Reviewer: ```suggestion
You can configure detailed instructions for any of the properties shown in the list by clicking the "Add" or "Edit" buttons. Once you have selected the type of instruction you wish to add and have filled in the required input fields, you are then able to click 'Add instruction' and return to the properties list. Following that, you can continue adding instructions for the remaining properties or you can submit your selection by clicking "Save and update instructions".

*Notes: instructions for required properties are added by default and can be edited but they cannot be deleted.*
```

---
title: "Managing Event Specifications in the Console"
sidebar_label: "Using the UI"
sidebar_position: 1
sidebar_custom_props:
  offerings:
    - enterprise
    - cloud
---

> ## Crafting a New Event Specification through the Console

Reviewer 2: Change to 
'Creating a new event specification through Console'

Author: done

Event specifications serve as direct counterparts to [data structures](/docs/understanding-tracking-design/managing-your-data-structures/ui/index.md) and encapsulate the documented events within a data product.

:::info
Please note that the creation of event specifications is exclusive to data products.
:::

> Create event specifications in tandem with the latest deployed data structure version in production. This ensures tracking implementation instructions align with the validation criteria in the event specification.

Reviewer 3: I thought we hadn't implemented that yet, but now I have some distant memory of James saying it's actually possible to use DS in dev? Would be good to confirm!

> Create event specifications in tandem with the latest deployed data structure version in production. This ensures tracking implementation instructions align with the validation criteria in the event specification.

Author: @jamessnowplow could you please confirm? thanks

> Create event specifications in tandem with the latest deployed data structure version in production. This ensures tracking implementation instructions align with the validation criteria in the event specification.

Author: ok discussed with james, changed to `development`

> Your event specification guarantees compatibility with the associated data structure version. This ensures events tracked using specified instructions pass validation for the associated data structure.

Reviewer 3: I feel the comment is accurate, it basically says (in my reading at least) that the instructions are compatible to the data structure and therefore it's not possible that you generate bad events if you follow them. Perhaps there can be some additional disambiguation.

> Your event specification guarantees compatibility with the associated data structure version. This ensures events tracked using specified instructions pass validation for the associated data structure.

Reviewer 3: ```suggestion
Your event specification guarantees compatibility with the associated data structure version, being a specialization of it. This ensures events tracked using specified instructions pass validation for the associated data structure.
```

> If a newer event specification version is deployed, consider upgrading event specification and tracking to the latest data structure, ensuring ongoing compliance with validation requirements.

Reviewer 3: My reading is an advice to always try to use the latest DS version if you're updating the event spec for whatever other reason. Would be good to rephrase if that's the case, but I'm not sure what would be a clear articulation either.

> If a newer event specification version is deployed, consider upgrading event specification and tracking to the latest data structure, ensuring ongoing compliance with validation requirements.

Author: I removed this text, I agree with you could be a bit confusing. IMO it is actually not necessary because of the third paragraph

## Creating and Editing Event Specifications

To create a new event specification, follow these steps:

1. Select a data product
2. Click the "Create event" button
3. A dialog will appear, prompting you to enter a name for your event specification and click "Save and continue"
4. Your first event specification will be displayed on the page

![Create an Event Specification](images/create-event-specification.png)

To add more information or modify an existing event specification, follow these steps:

1. Navigate to the appropriate data product
2. Select the desired event specification
3. This action will open on overview of the selected event specification containing the details that have been added to date.

This interface is divided into focused sections; explore each section below for more details.

![Example of an Event Specification Overview](images/event-specification-overview.png)

### Event Information

> This section provides essential meta-information for your event specification, including the event name, description, and the applications in which this event is tracked.

Reviewer 2: Provide essential meta-information for your event specification, including a descriptive name, a description of the event, and the applications on which the event is tracked.

### Event Data Structure

> This section defines the structure of your data and specifies the types of properties associated with this event, along with its version.

Author: done

You can choose from two types of data structures:

- **Standard**; Provided by the Snowplow tracker by default
- **Custom**; Provided by your organization

![Event Data Structure](images/event-data-structure.png)

## Entity Data Structures

> Within this section, you have the flexibility to choose the specific entities that should be associated with the event when it is triggered.

Author: done thanks

> To facilitate making an informed selection of your entities, you are able to view a detailed breakdown of the properties associated with your chosen entity for a specific version.

Reviewer 2: You can view a detailed breakdown of the properties associated with the chosen entities (and its version) by ....

Author: done

![Entity Data Structures](images/entity-data-structures.png)

### Event Triggers

This section defines the locations and circumstances under which this event is triggered.

To create your first trigger, simply select the "Add trigger" button. You can edit or delete an existing trigger by clicking the dedicated buttons beside each entry in the triggers list.

A dialog will appear, allowing you to upload an image and provide additional context, such as the URL to the page on which this trigger applies.

![Event Triggers](images/event-triggers.png)

### Properties

This section allows you to specify how each property for a selected event data structure should be populated.

> The dialog displays the list of properties defined in the data structure associated with this event specification.

Author: done thanks

> You can configure detailed instructions for any of the properties shown in the list by clicking the "Add" or "Edit" buttons. Once you have selected the type of instruction you wish to add and have filled in the required input fields, you are then able to click 'Add instruction' and return to the properties list. Following that, you can continue adding instructions for the remaining properties or you can submit your selection by clicking "Save and update instructions".

Author: removed last bit and revised text
